topics/lua-api.md

```diff
@@ -411,21 +405,32 @@ The named argument version except the following named arguments:
 * _flags_ - Lua array of strings representing function flags
 * _description_ - Lua string representing the function description
 
-_function\_name_ and _callback_ are mandatory, the other named arguments are option. Error raise on the following cases:
-
-* Missing mandatory arguments
-* Unkown named argument
-* Wrong argument type
-* Unknows flag
-* Another function with the same name already registered on the current library
+_function\_name_ and _callback_ are mandatory, the other named arguments are option.
 
 Usage Example:
 
 ```
 redis> FUNCTION LOAD Lua mylib "redis.register_function{function_name='noop', callback=function() end, function, flags={'no-writes'}, description='some desc')"
 ```
 
-For more information about function flags refer to [Function Flags](functions-intro#function-flags)
+#### <a name="script_flags"></a> Script Flags
+
+The following is a list of supported script flags:
+
+* `no-writes` - indicating the function perform no writes which means that it is OK to run it on:
+   * read-only replica
+   * Using FCALL_RO
+   * If disc error detected
+   
+   It will not be possible to run a function in those situations unless the function turns on the `no-writes` flag, in this case Redis will inforce the function is doing not writes and will raise an error if it does.
+
+* `allow-oom` - indicate that its OK to run the function even if Redis is in OOM (out of memory) state, if the function will not turn on this flag it will not be possible to run it if OOM reached (even if the function declares `no-writes` and even if `fcall_ro` is used). If this flag is set, any command will be allowed on OOM (even ones that are normally not allowed in that state).
```

Comment:
```suggestion
* `allow-oom` - indicate that it's OK to run the script even if Redis is in OOM (out of memory) state, if the script will not turn on this flag it will not be possible to run it if OOM reached (even if the script declares `no-writes` and even if `fcall_ro` / `eval_ro` is used). If this flag is set, any command will be allowed on OOM (even ones that are normally not allowed in that state).
```